src/shared/log_builder.c

```diff
@@ -241,49 +243,50 @@ int log_builder_update_hostname(log_builder_t * builder) {
 
 // Update the host's IP value
 int log_builder_update_host_ip(log_builder_t * builder) {
-    char * host_ip = NULL;
+    static char host_ip[IPSIZE] = { '\0' };
+    static time_t last_update = 0;
+    time_t now = time(NULL);
 
+    if ((now - last_update) > ip_update_interval) {
```

Comment:
`ip_update_interval` should be less or equal than (now-last_update) to be accurate. Currently IP address is updated every `ip_update_interval` + 1
```suggestion
    if ((now - last_update) >= ip_update_interval) {
```